odoo/tools/pdf.py

```diff
@@ -6,15 +6,34 @@
 from datetime import datetime
 from hashlib import md5
 from logging import getLogger
-from PyPDF2 import PdfFileWriter, PdfFileReader
-from PyPDF2.generic import DictionaryObject, NameObject, ArrayObject, DecodedStreamObject, NumberObject, createStringObject, ByteStringObject
 from zlib import compress, decompress
 from PIL import Image
 from reportlab.lib import colors
 from reportlab.lib.units import cm
 from reportlab.lib.utils import ImageReader
 from reportlab.pdfgen import canvas
 
+# Patch PyPDF2 for compatibility between < 2.0 >
+# https://pypdf2.readthedocs.io/en/latest/user/migration-1-to-2.html
+try:
+    from PyPDF2 import PdfFileWriter as PdfWriter, PdfFileReader as PdfReader, PdfFileMerger as PdfMerger
+    from PyPDF2.utils import PdfReadError
+
+    PdfWriter._add_object = PdfWriter._addObject
+
+    # by default PdfFileReader will overwrite warnings.showwarning which is what
+    # logging.captureWarnings does, meaning it essentially reverts captureWarnings
+    # every time it's called which is undesirable
+    # The overwrite was removed in PyPDF2 > 2.0 see https://github.com/py-pdf/PyPDF2/pull/225
+    old_init = PdfReader.__init__
+    PdfReader.__init__ = lambda self, stream, strict=True, warndest=None, overwriteWarnings=True: \
+    old_init(self, stream=stream, strict=strict, warndest=None, overwriteWarnings=False)
```

Comment:
if we don't use PdfReader anywhere but only OdooPdfReader, we could overwrite that in OdooPdfReader

Reply:
seems a bit tricky as this is only needed for PyPDF2 >= 2.0